packages/components/src/button/test/index.tsx

```diff
@@ -220,6 +220,100 @@ describe( 'Button', () => {
 			).toBeVisible();
 		} );
 
+		describe( 'when isPressed is set', () => {
+			it( 'should set aria-pressed', () => {
+				render( <Button isPressed /> );
+
+				expect( screen.getByRole( 'button' ) ).toHaveAttribute(
+					'aria-pressed',
+					'true'
+				);
+			} );
+
+			it( 'should use given aria-pressed value if provided', () => {
+				render( <Button isPressed aria-pressed={ false } /> );
+
+				expect( screen.getByRole( 'button' ) ).toHaveAttribute(
+					'aria-pressed',
+					'false'
+				);
+			} );
+
+			it( 'should use given aria-checked value if provided', () => {
+				render(
+					<Button isPressed role="checkbox" aria-checked={ false } />
+				);
+
+				expect( screen.getByRole( 'checkbox' ) ).not.toBeChecked();
+			} );
+
+			it( 'should use given aria-selected value if provided', () => {
+				render(
+					<Button isPressed role="option" aria-selected={ false } />
```

Comment:
Are there scenarios where `isPressed` is `true` with `aria-selected` and `aria-checked` set to `false` for roles like `option` and `checkbox`? I'm curious!

This also made me wonder if we could check the value of `aria-selected`, `aria-checked` , etc. via `userEvent` to test closer to how it'll be used in the DOM. 🤔